src/arrayMethodSort.js

```diff
@@ -5,7 +5,34 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let count = true;
+
+    do {
+      count = false;
+
+      for (let i = 1; i < this.length; i++) {
+        const prev = this[i - 1];
+        const current = this[i];
+
+        if (compareFunction) {
+          if (compareFunction(prev, current) > 0) {
+            count = true;
+            this[i - 1] = current;
+            this[i] = prev;
+          }
+        }
+
+        if (!compareFunction) {
+          if (`${prev}` > `${current}`) {
+            count = true;
+            this[i - 1] = current;
+            this[i] = prev;
+          }
+        }
```

Comment:
Create separate `compareFunction` that will be used in case when no compare function will used